src/Discovery/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+
+import Discovery from './Discovery';
+import { DiscoveryConfig } from './DiscoveryConfig';
+import { userHasMethodForServiceOnResource } from '../authMappingUtils';
+import { hostname, discoveryConfig, useArboristUI } from '../localconf';
+
+const loadStudiesFromMDS = async (): Promise<any[]> => {
+  // Why `_guid_type='discovery_metadata'? We need to distinguish the discovery page studies in MDS
+  // from other data in MDS. So all MDS records with `_guid_type='discovery_metadata'` should be
+  // the full list of studies for this commons.
+  const GUID_TYPE = 'discovery_metadata';
+  const LIMIT = 1000; // required or else mds defaults to returning 10 records
+  const MDS_URL = `${hostname}mds/metadata`;
+  const STUDY_DATA_FIELD = 'gen3_discovery'; // field in the MDS response that contains the study data
+
+  try {
+    let allStudies = [];
+    let offset = 0;
+    // request up to LIMIT studies from MDS at a time.
+    let shouldContinue = true;
+    while (shouldContinue) {
+      const url = `${MDS_URL}?data=True&_guid_type=${GUID_TYPE}&limit=${LIMIT}&offset=${offset}`;
+      // It's OK to disable no-await-in-loop rule here -- it's telling us to refactor
+      // using Promise.all() so that we can fire multiple requests at one.
+      // But we WANT to delay sending the next request to MDS until we know we need it.
+      // eslint-disable-next-line no-await-in-loop
+      const res = await fetch(url);
+      if (res.status !== 200) {
+        throw new Error(`Request for study data at ${url} failed. Response: ${JSON.stringify(res, null, 2)}`);
+      }
+      // eslint-disable-next-line no-await-in-loop
+      const jsonResponse = await res.json();
+      const studies = Object.values(jsonResponse).map(entry => entry[STUDY_DATA_FIELD]);
+      allStudies = allStudies.concat(studies);
+      const noMoreStudiesToLoad = studies.length < LIMIT;
```

Comment:
Just want to make sure a potential edge case is covered in here: what if the MDS has exactly 1000 records? Would the 2nd request error out or finishes with 200 but with no data being returned?

Reply:
Great q, let me test that!

Reply:
I just tested this by setting LIMIT=137 which results in 2 queries, one with LIMIT=137 OFFSET=0 which returns all 137 studies, and then a second query which returns no studies (`{}`). The empty object doesn't cause any problems, it results in a empty list of studies which gets concat'ed with `allStudies` so the page works like normal. Thanks for catching this edge case!

Reply:
yay awesome! thanks for checking